pkg/egressgateway/manager.go

```diff
@@ -57,7 +58,13 @@ func NewEgressGatewayManager(k8sCacheSyncedChecker k8sCacheSyncedChecker) *Manag
 // sync with k8s and then runs the first reconciliation.
 func (manager *Manager) runReconciliationAfterK8sSync() {
 	go func() {
-		manager.k8sCacheSyncedChecker.WaitUntilK8sCacheIsSynced()
+		for {
+			if manager.k8sCacheSyncedChecker.K8sCacheIsSynced() {
+				break
+			}
+
```

Comment:
non-blocking comment: If there is a bug that does not allow the cache to be synced, we would get stuck in an infinite loop. Should we add an info message here? We can print it after N iterations if we think it would be too verbose.

Reply:
Looks like we are already logging this info in the `watchers` package: https://github.com/cilium/cilium/blob/25810847a11b659df9056a32b907e0da62fd5666/pkg/k8s/watchers/watcher.go#L391-L398

so probably no need to log it also here